sdk/src/main/java/io/opentelemetry/sdk/metrics/data/MetricData.java

```diff
@@ -152,6 +157,64 @@ public static DoublePoint create(
     }
   }
 
+  @Immutable
+  @AutoValue
+  public abstract static class LongSummaryPoint extends Point {
+
+    LongSummaryPoint() {}
+
+    public abstract long getCount();
+
+    public abstract long getSum();
+
+    @Nullable
+    public abstract Long getMin();
+
+    @Nullable
+    public abstract Long getMax();
```

Comment:
you mean in the naming of the methods, or something more than that?

Reply:
My goal was to have it match the MinMaxSumCount aggregation mentioned in here:  https://github.com/open-telemetry/opentelemetry-specification/blob/2b75442b05fd968f197422dc18124338a955f3a2/specification/sdk-metric.md#aggregator-implementations

I see that that description also has some surprising requirements around atomicity that should probably be discussed.

Reply:
The aggregation should match the data you produce, but the way we export data in MetricData should match the proto definition to be easier to consume.

Reply:
So what are you suggesting we change these methods to?

Reply:
I mean you should have a `public List<ValueAtPercentile> getPercentileValues` where `ValueAtPercentile` is a class that contains the percentile + value. In case of MinMaxSumCount only 2 percentiles will be available.

Reply:
If we want to have the types and points match with the proto, otherwise we can have MinMaxCountSum specialized points that we document how to transform them in the Summary proto points.

Reply:
ah, I see! Yeah, I guess that makes sense. Do we think we'll use the summary for other aggregations?

Reply:
Yes, I guess so.

Reply:
ok, I'll switch it to have that API

Reply:
done, please TAL

Reply:
I'm having some concerns about the current API for use in exporters, @bogdandrutu. It's important (at least for NR) to know that a point is a MinMaxSumCount-generated datapoint vs. some other summarization because exporters will handle these differently.  Also, I need to know if a LongPoint came from a counter vs. coming from a observer gauge because the backend handles these differently. Is there a way we can attach the aggregation type to the Point/MetricData, so that exporters will know how to handle them appropriately?

Reply:
Should the `Type` be much more granular and describe the type of aggregation, rather than the type of output?

Reply:
I am not sure I understand, what is the difference that you have in mind?
> Also, I need to know if a LongPoint came from a counter vs. coming from a observer gauge because the backend handles these differently. Is there a way we can attach the aggregation type to the Point/MetricData, so that exporters will know how to handle them appropriately?

For the MinMaxSumCount vs Summary for export can you clarify what would be different?

Reply:
In New Relic, we have a native metric for MinMaxSumCount summaries, so we can create those directly if the data is of that shape. For higher order summaries (more percentiles than just 0 and 100), we currently export them as a set of metrics that represent each of the percentile values. So, knowing which is which is important to know which path to go down. 

With the current model, we could inspect the percentiles and see if they're only 0 and 100, but that feels pretty clunky as opposed to just knowing that a point was aggregated directly into a MinMaxSumCount.